### ISSUE TITLE
[LifetimeSafety] False positive after adding to an std::vector

### ISSUE THREAD

**Issue by mphe** [llvm/llvm-project#222537]
The following code produces a false positive lifetime warning about `vec` being invalidated after `push_back()`. Same goes for `emplace_back()`.
```cpp
void repro(std::vector<int>& vec)
{
    vec.push_back(1);
    vec[0] = 2;
}
```

Compiled with `-std=c++20 -Wlifetime-safety-invalidation`

```
<source>:3:12: warning: parameter 'vec' is later invalidated [-Wlifetime-safety-invalidation]
    3 | void repro(std::vector<int>& vec)
      |            ^~~~~~~~~~~~~~~~~~~~~
<source>:5:9: note: parameter 'vec' is invalidated here
    5 |     vec.push_back(1);
      |     ~~~~^~~~~~~~~~~~
<source>:6:5: note: later used here
    6 |     vec[0] = 2;
      |     ^~~
```

Here is the code on godbolt: https://godbolt.org/z/xvrfo7rj6
This issue appeared with 23.1.0 and still exists in 23.1.1.

```
clang version 23.1.1 (https://github.com/llvm/llvm-project 6dfe1677ab8dffbc6ec13d53a1e0215d75147689)
Target: x86_64-pc-windows-msvc
Thread model: posix
Build config: +alloc:rpmalloc
```

**Comment by usx95** [comment]
Ack. Invalidations has a lot of false-positives right now and not adoptable atm.

FWIW, this would be fixed if we start having projections in paths and invalidations talk about interior paths instead of complete paths with is status quo.
